azure-secret-manager/src/main/java/io/micronaut/azure/secretmanager/AzureVaultConfigurationClient.java

```diff
@@ -0,0 +1,129 @@
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.env.PropertySource;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.discovery.config.ConfigurationClient;
+import io.micronaut.scheduling.TaskExecutors;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Distributed configuration client implementation that fetches application secret values from Azure keyvalut.
+ */
+@Singleton
+@Requires(beans = SecretClient.class)
+@BootstrapContextCompatible
+public class AzureVaultConfigurationClient implements ConfigurationClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AzureVaultConfigurationClient.class);
+
+    private final AzureKeyvaultConfigurationProperties azureKeyvaultConfigurationProperties;
+    private final ExecutorService executorService;
+    private final SecretKeyvaultClient secretClient;
+
+    /**
+     * Default Constructor.
+     *
+     * @param azureKeyvaultConfigurationProperties Azure Secret Vault Client Configuration
+     * @param executorService                      Executor Service
+     * @param secretClient                         The secrets client
+     */
+    public AzureVaultConfigurationClient(
+            AzureKeyvaultConfigurationProperties azureKeyvaultConfigurationProperties,
+            @Named(TaskExecutors.IO) @Nullable ExecutorService executorService,
+            SecretKeyvaultClient secretClient) {
+        this.azureKeyvaultConfigurationProperties = azureKeyvaultConfigurationProperties;
+        this.executorService = executorService;
+        this.secretClient = secretClient;
+    }
+
+    @Override
+    public Publisher<PropertySource> getPropertySources(Environment environment) {
+
+        if (azureKeyvaultConfigurationProperties.getVaultURL() == null || azureKeyvaultConfigurationProperties.getVaultURL().equals("")) {
+            return Flux.empty();
+        }
+
+        List<Flux<PropertySource>> propertySources = new ArrayList<>();
+        Scheduler scheduler = executorService != null ? Schedulers.fromExecutor(executorService) : null;
+
+        Map<String, Object> secrets = new HashMap<>();
+
+        int retrieved = 0;
+
+        for (VersionedSecret versionedSecret : secretClient.listSecrets()) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Retrieving secrets from Azure Secret Vault with URL: {}", azureKeyvaultConfigurationProperties.getVaultURL());
```

Comment:
If I understand correctly, if there will be 10 secrets, it will generate 10 log messages with same content. If so I suggest to place this LOG block above the for loop.